docs/guide/interactions/slash_commands.rst

```diff
@@ -0,0 +1,1618 @@
+
+            now = discord.utils.utcnow().date() # only get the `date` component from the datetime
+            if not self._allow_past_dates and when < now:
+                raise app_commands.AppCommandError('this date is in the past!')
+                # note: more on exceptions and error handling later in this page...
+
+            return when
+
+    # to disallow past dates when annotating:
+    Transform[datetime.datetime, DateTransformer(past=False)]
+
+Since the parameter's type annotation is replaced with :class:`~.app_commands.Transform`,
+the underlying Discord type and other information must now be provided
+through the :class:`~.app_commands.Transformer` itself.
+
+These can be provided by overriding the following properties:
+
+- :attr:`.Transformer.type`
+- :attr:`.Transformer.min_value`
+- :attr:`.Transformer.max_value`
+- :attr:`.Transformer.choices`
+- :attr:`.Transformer.channel_types`
+
+Since these are properties, they must be decorated with :class:`property`:
+
+.. code-block:: python
+
+    class UserAvatar(app_commands.Transformer):
+        async def transform(self, interaction: discord.Interaction, user: discord.User) -> discord.Asset:
+            return user.display_avatar
+
+        # changes the underlying type to discord.User
+        @property
+        def type(self) -> discord.AppCommandOptionType:
+            return discord.AppCommandOptionType.user
+
+.. (todo) talk about this properly and write an example
+
+:meth:`~.Transformer.autocomplete` callbacks can also be defined in-line.
+
+.. _guide_slash_commands_type_conversion:
+
+Type conversion
+++++++++++++++++
+
+The table below outlines the relationship between Discord and Python types.
+
++-----------------+------------------------------------------------------------------------------------+
+|   Discord Type  |                                Python Type                                         |
++=================+====================================================================================+
+| ``STRING``      | :class:`str`                                                                       |
++-----------------+------------------------------------------------------------------------------------+
+| ``INTEGER``     | :class:`int`                                                                       |
++-----------------+------------------------------------------------------------------------------------+
+| ``BOOLEAN``     | :class:`bool`                                                                      |
++-----------------+------------------------------------------------------------------------------------+
+| ``NUMBER``      | :class:`float`                                                                     |
++-----------------+------------------------------------------------------------------------------------+
+| ``USER``        | :class:`~discord.User` or :class:`~discord.Member`                                 |
```

Comment:
I feel like it's worth noting that when using `discord.Member` as a typehint, that it will not have any presence data (e.g. there is no data on the online status of an member)

Reply:
Thinking about it, you are right about it